docs/protocol/extension_logging.md

```diff
@@ -0,0 +1,199 @@
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+-->
+
+# Logging Extension
+
+This document describes Triton's logging extension. The logging extension enables
+the client to configure log settings during a Triton run. Triton reports "logging" 
+in the extensions field of its Server Metadata.
+
+## HTTP/REST
+
+In all JSON schemas shown in this document $number, $string, $boolean,
+$object and $array refer to the fundamental JSON types. #optional
+indicates an optional JSON field.
+
+Triton exposes the logging endpoint at the following URL. The client may use
+HTTP GET request to retrieve the current log settings. A HTTP POST request
+will modify the log settings, and the endpoint will return the updated log
+settings on success or an error in the case of failure. 
+
+```
+GET v2/logging
+
+POST v2/logging
+```
+
+### Log Setting Response JSON Object
+
+A successful log setting request is indicated by a 200 HTTP status
+code. The response object, identified as $log_setting_response, is
+returned in the HTTP body for every successful log setting request.
+
+```
+$log_setting_response =
+{
+  $log_setting, ...
+}
+
+$log_setting = $string : $string | $boolean | $number
+```
+
+Each $log_setting JSON describes a “name”/”value” pair, where the “name” is
+the $string representation of the log setting and the “value” is a $string, $bool, or $number representation of the
+setting value. Currently, the following log settings are defined:
+
+- "log_file" : a $string parameter defining the file where the log outputs will be saved. If not specified,
+log outputs will stream to the console.
```

Comment:
"If not specified" -> "If empty string is specified". I think "not specified" just means that the user doesn't intend to change current setting.